### PR TITLE
Attach fallback policies to sites

### PR DESCRIPTION
**Pull request by C-gyorfi**
## Context

- attach fallback policy to site, and make sure it is a single select dropdown
- update list of attached policies to include fallback policy
- update select dropdown to show selected attached fallback policy

### CODE REVIEW

**Review by yusufsheiqh — APPROVED** [review]

File: app/controllers/sites_controller.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 class SitesController < ApplicationController
-  before_action :set_site, only: %i[show edit update destroy policies attach_policies]
+  before_action :site, only: %i[show edit update destroy policies attach_policies]
```
Comment (emileswarts) [inline app/controllers/sites_controller.rb:2]: Preferred the verb "set_site", not sure "site" is more descriptive

File: app/controllers/sites_controller.rb [hunk 2]
```diff
@@ -89,11 +91,17 @@ def site_id
     params.fetch(:id)
   end
 
-  def set_site
-    @site = Site.find(site_id)
+  def site
+    @site ||= Site.find(site_id)
```
Comment (emileswarts) [inline app/controllers/sites_controller.rb:95]: don't need to memoize

File: app/controllers/sites_controller.rb [hunk 3]
```diff
@@ -55,7 +55,10 @@ def destroy
   end
 
   def policies
-    @policies = Policy.all
+    default_fallback_policy = OpenStruct.new(name: "No fallback policy")
```
Comment (emileswarts) [inline app/controllers/sites_controller.rb:58]: Do we need to set this here or can we push it into the view as a placeholder

File: app/controllers/sites_controller.rb [hunk 4]
```diff
@@ -55,7 +55,10 @@ def destroy
...
+    @fallback_policies = Policy.where(fallback: true).to_a.unshift default_fallback_policy
```
Comment (emileswarts) [inline app/controllers/sites_controller.rb:59]: consider pushing the where into a helper method like Policy.fallback_policies

File: app/views/sites/policies.html.erb [hunk 5]
```diff
@@ -1,15 +1,24 @@
...
-
 <%= form_with local: true do |f| %>
   <div class="govuk-form-group">
-    <fieldset class="govuk-fieldset" aria-describedby="contact-hint">
-      <div id="contact-hint" class="govuk-hint">
-        Select all policies to attach to this site.
+    <div id="attach-policies-hint" class="govuk-hint">
+        Select a fallback policy to attach to this site.
+    </div>
+    <% selected_fallback_policy = @site.policies.select(&:fallback).first %>
```
Comment (emileswarts) [inline app/views/sites/policies.html.erb:7]: @site.fallback_policy

File: app/views/sites/show.html.erb [hunk 6]
```diff
@@ -20,11 +20,21 @@
   <%= link_to "Add client", new_site_client_path(site_id: @site), class: "govuk-button" %>
 <% end %>
 
-<% if @site.policies.any? %>
+<h3 class="govuk-heading-m">List of attached policies</h3>
+<p class="govuk-body">
+  <span class="govuk-!-font-weight-bold">Fallback policy:</span>
+  <% if @site.policies.select(&:fallback).first %>
+    <%= @site.policies.select(&:fallback).first.name %>
```
Comment (emileswarts) [inline app/views/sites/show.html.erb:27]: @site.policies.select(&:fallback).first.try(:name) || "There are no fallback policies attached to this site."

File: config/routes.rb [hunk 7]
```diff
@@ -13,7 +13,7 @@
   end
 
   get "/sites/:id/policies", to: "sites#policies", as: "site_policies"
-  post "/sites/:id/policies", to: "sites#attach_policies", as: "attach_site_policies"
+  post "/sites/:id/policies", to: "sites#attach_policies"
```
Comment (emileswarts) [inline config/routes.rb:16]: try stick with Restful actions

**Pull request merged**